doc/contributions/introduction.md

```diff
@@ -0,0 +1,80 @@
+# Introduction
+
+## Git and Github
+
+We recommend you to get familiar with the
+[git](https://guides.github.com/introduction/git-handbook) and
+[Github](https://guides.github.com) workflows before posting any changes.
+
+### Set up in a nut shell
+
+The following steps describe the process in a nut shell to provide you a basic
+template:
+
+* Create an account on [GitHub](https://github.com)
+* Fork the [shadow-utils repository](https://github.com/shadow-maint/shadow)
+* Clone the SSSD repository
+
+```
+git clone git@github.com:shadow-maint/shadow.git
+```
+
+* Add your fork as an extra remote
+
+```
+git remote add $ghusername git@github.com:$ghusername/shadow.git
+```
+
+* Setup your name contact e-mail that you want to use for the development
+
+```
+git config user.name "John Smith"
+git config user.email "john.smith@home.com"
+```
+
+**Note**: this will setup the user information only for this repository. You
+can also add `--global` switch to the `git config` command to setup these
+options globally and thus making them available in every git repository.
+
+* Create a working branch
+
+```
+git checkout -b my-changes
+```
+
+* Commit changes
+
+```
+vim change-what-you-need
+git commit -s
+```
+
+* Push your changes to your GitHub repository
+
+```
+git push $ghusername my-changes --force
+```
+
+* Open a Pull Request against SSSD project (using `gh` or web ui)
+
+```
+gh pr create
+```
+
+* Make sure that all Continuous Integration checks are green and wait review
+
+## Mailing patches
+
+TODO: write
```

Comment:
I think for now it's ok to say something like "the github pull request workflow is the only currently supported patch submission mechanism".

If people actually want it, we can pursue a shadow@vger.kernel.org or somesuch.  If we move to sr.ht, then we get a mailing list for free there.  I *could* run a list, but I'd rather not deal with the inevitable abuse.

Reply:
The "bye bye github" issue is a bit too all-encompassing for quick action to be expected.

So I suggest saying here "no mailing list currently exists", then opening an issue for creating a mailing list.  We can hash out there where to put it.